viewer/packages/tools/src/Timeline/TimelineTool.ts

```diff
@@ -0,0 +1,203 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import TWEEN from '@tweenjs/tween.js';
+
+import { Cognite3DModel } from '@reveal/core';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import { Keyframe } from './Keyframe';
+import { TimelineDateUpdateDelegate } from './types';
+import { EventTrigger, assertNever } from '@reveal/core/utilities';
+
+/**
+ * Tool to applying styles to nodes based on date to play them over in Timeline
+ */
+export class TimelineTool extends Cognite3DViewerToolBase {
+  private readonly _model: Cognite3DModel;
+  private _keyframes: Keyframe[];
+  private _playback: TWEEN.Tween | undefined = undefined;
+  private readonly _events = { dateChanged: new EventTrigger<TimelineDateUpdateDelegate>() };
+
+  constructor(cadModel: Cognite3DModel) {
+    super();
+
+    this._model = cadModel;
+    this._keyframes = new Array<Keyframe>();
+  }
+
+  /**
+   * Subscribe to the Date changed event
+   * @param event `dateChanged` event
+   * @param listener Listen to Timeline date Update during Playback
+   */
+  public subscribe(event: 'dateChanged', listener: TimelineDateUpdateDelegate): void {
+    switch (event) {
+      case 'dateChanged':
+        this._events.dateChanged.subscribe(listener);
+        break;
+      default:
+        assertNever(event, `Unsupported event: '${event}'`);
+    }
+  }
+
+  /**
+   * Unsubscribe to the Date changed event
+   * @param event `dateChanged` event
+   * @param listener Remove Listen to Timeline date Update
+   */
+  public unsubscribe(event: 'dateChanged', listener: TimelineDateUpdateDelegate): void {
+    switch (event) {
+      case 'dateChanged':
+        this._events.dateChanged.unsubscribe(listener as TimelineDateUpdateDelegate);
```

Comment:
```suggestion
        this._events.dateChanged.unsubscribe(listener);
```
No need to cast, already correct type